pipelines/zendesk/__init__.py

```
@@ -0,0 +1,221 @@
+    """
+    all_metric_events = zendesk_client.ticket_metric_events(start_time=start_time)
+    for metric_event in all_metric_events:
+        yield metric_event.to_dict()
+
+
+@dlt.resource(name="chats", write_disposition="replace")
+def chats_table(zendesk_client: Zenpy, start_date: DateTime = FIRST_DAY_OF_CURRENT_YEAR) -> Iterator[TDataItem]:
+    """
+    Resource for Chats
+    @:param zendesk_client: Zenpy type object, used to make calls to Zendesk API through zenpy module
+    @:param start_date: Datetime object that serves as the starting date of when to retrieve objects
+    @:returns: Generator returning dicts for every row of data
+    """
+
+    all_chats = zendesk_client.chats.incremental(start_time=start_date)
+    dict_generator = basic_load(resource_api=all_chats)
+    yield from dict_generator
+
+
+def basic_resource(zendesk_client: Zenpy, resource: str, per_page: int = 1000) -> Iterator[TDataItem]:
+    """
+    Basic loader for most endpoints offered by Zenpy. Supports pagination. Expects to be called as a DLT Resource
+    @:param zendesk_client: Zenpy type object, used to make calls to Zendesk API through zenpy module
+    @:param resource: String that references the zenpy endpoint, usually directly linked to a ZendeskAPI endpoint, ex: users
+    @:param per_page: Indicated how many resources should be retrieved per page with the default being 1000
+    @:returns: Generator of dicts
+    """
+
+    try:
+        # use the zenpy endpoint with methods to yield the data
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api = resource_api_method()[::per_page]
+    except TypeError:
+        # this resource doesn't support slicing for pagination
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api = resource_api_method()
+    except Exception as e:
+        logger.warning(f"Skipped resource: {resource}. Couldn't load!")
```

swallowing exceptions looks really dangerous. you should log and then raise

---

This was a temporary fix to make the resources failure resilient, since we don't want the pipeline to completely fail when a single resource fails.

---

we are working on requests drop in replacement that will do that for you

---

Ok, then I shall just remove this part in and similar parts in resource code.